lib/Net/Kubernetes/Resource/Deployment.pm

```diff
@@ -0,0 +1,27 @@
+package Net::Kubernetes::Resource::Deployment;
+
+use Moose;
+
+extends 'Net::Kubernetes::Resource';
+with 'Net::Kubernetes::Resource::Role::State';
+with 'Net::Kubernetes::Resource::Role::Spec';
+with 'Net::Kubernetes::Resource::Role::HasPods';
+with 'Net::Kubernetes::Resource::Role::HasReplicaSets';
+
+sub scale {
+    my($self, $replicas, $timeout) = @_;
+    $timeout ||= 5;
```

Comment:
Likely the best way to do that is to put an attribute in the APIAccess role and advertise it for the constructor of Net::Kubernetes. 

I like 0 for return immediately and -1 for wait forever.